File: sensor-data-java/src/main/java/pipelines/examples/sensordata/SensorDataIngress.java
package pipelines.examples.sensordata;

import pipelines.akkastream.AkkaServerStreamlet;

import pipelines.akkastream.StreamletLogic;
import pipelines.akkastream.util.javadsl.HttpServerLogic;

import pipelines.streamlets.RoundRobinPartitioner;
import pipelines.streamlets.StreamletShape;
import pipelines.streamlets.avro.AvroOutlet;

import akka.http.javadsl.marshallers.jackson.Jackson;

public class SensorDataIngress extends AkkaServerStreamlet {
  AvroOutlet<SensorData> out =  AvroOutlet.<SensorData>create("out", SensorData.class)
          .withPartitioner(RoundRobinPartitioner.getInstance());

  public StreamletShape shape() {
   return StreamletShape.createWithOutlets(out);
  }

  public StreamletLogic createLogic() {
    return HttpServerLogic.createDefault(this, out, Jackson.byteStringUnmarshaller(SensorData.class), getStreamletContext());
  }
}

File: sensor-data-scala/src/main/scala/pipelines/examples/sensordata/SensorDataIngress.scala
package pipelines.examples.sensordata

import akka.http.scaladsl.marshallers.sprayjson.SprayJsonSupport._

import pipelines.akkastream.AkkaServerStreamlet
import pipelines.akkastream.util.scaladsl._
import pipelines.streamlets.StreamletShape
import pipelines.streamlets.avro._

import SensorDataJsonSupport._

object SensorDataIngress extends AkkaServerStreamlet {
  val out = AvroOutlet[SensorData]("out", s ⇒ s.deviceId.toString + s.timestamp.toString)
  def shape = StreamletShape.withOutlets(out)
  override def createLogic = HttpServerLogic.default(this, out)
}
